tests/test_meta/test_grouped_transformer.py

```diff
@@ -408,3 +410,32 @@ def test_set_output(penguins_df, frame_func, transform_output):
     meta = GroupedTransformer(StandardScaler(), groups="island").set_output(transform=transform_output)
     transformed = meta.fit_transform(X, y)
     assert isinstance(transformed, frame_func)
+
+def test_with_object_dtype():
+    # Example from https://github.com/koaning/scikit-lego/issues/740
+
+    set_config(transform_output="pandas")
+
+    data = {
+        "big": ["A", "A", "A", "A", "A", "B", "B", "B", "C", "C"],
+        "small": ["a", "a", None, "a", "a", "b", "b", None, "C", "C"],
+        "other": [0.1, 0.2, 0.3, 0.6, 0.5, 0.1, 0.3, 0.5, 0.6, 0.6],
+        "y": [1, 1, 0, 1, 0, 1, 1, 0, 0, 0],
+    }
+    df = pd.DataFrame(data=data)
+    X, y = df.drop(columns=["y"]), df["y"]
+
+    result = GroupedTransformer(
+        transformer=SimpleImputer(strategy="most_frequent", missing_values=None),
+        groups=["big"],
+        check_X=False,
+    ).fit_transform(X, y)
+
+    expected = pd.DataFrame(
+        {
+            "small": ["a", "a", "a", "a", "a", "b", "b", "b", "C", "C"],
+            "other": [0.1, 0.2, 0.3, 0.6, 0.5, 0.1, 0.3, 0.5, 0.6, 0.6],
+        }
+    ).astype("object")
+
+    assert_frame_equal(result, expected)
```

Comment:
@kamilm moving our conversation here - This is a quick patch to allow _any_ return type, yet you might end up with a situation like this one, namely an numeric converted to object.

From my point of view it's not an issue, yet it's delegating responsibility to the user to convert `"other"` back to numeric